### PR TITLE
Driving Smarts 2023 environments

### PR DESCRIPTION
**Pull request by Adaickalavan**
+ Added new driving-smarts-v2023 benchmark
+ Added new platoon-v0 env
+ Modified naming of benchmark used in NeurIPS 2022 from driving-smarts-competition-env to driving-smarts-v2022

### CODE REVIEW

**Comment by qianyi-sun** [comment]
since we havent officially announced this, would it better if we name the PR and the env name differently until we launch this later?

**Comment by Adaickalavan** [comment]
> since we havent officially announced this, would it better if we name the PR and the env name differently until we launch this later?

This is simply a new set of enviroments which we are adding into SMARTS. It could be used for any purpose. I guess it is better to simply add it in now rather than later. Users are free to explore and start early.

**Review by saulfield — APPROVED** [review]

File: smarts/benchmark/driving_smarts/v2023/config.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# smarts/benchmark/driving_smarts_v2023/config.yaml
+---
+benchmark:
+  name: "Driving SMARTS 2023"
+  message: | # A useful message given at the start of the benchmark.
+    This is the `Driving SMARTS` 2023 benchmark.
+    For context see: 
+        - https://codalab.lisn.upsaclay.fr/competitions/
+        - https://smarts-project.github.io/archive/
```
Comment (Gamenot) [inline smarts/benchmark/driving_smarts/v2023/config.yaml:9]: Archive is not relevant here.

File: smarts/benchmark/driving_smarts/v2022/config.yaml [hunk 2]
```diff
@@ -31,16 +31,16 @@ benchmark:
...
+    #   loc: "smarts.env:driving-smarts-bubble-v2022"
+    #   scenarios:
+    #     - 6
+    #     - 4
+    #     - 42
     #   kwargs:
-    # naturalistic: # reserved for driving_smarts_competition_naturalistic-v0
-    #   loc: "smarts.env:driving_smarts_competition_naturalistic-v0"
+    # naturalistic:
+    #   loc: "smarts.env:driving-smarts-naturalistic-v2022"
```
Comment (Gamenot) [inline smarts/benchmark/driving_smarts/v2022/config.yaml:42]: This reminds me that we should add a standard way to deal with the naturalistic scenarios.

**Review by Gamenot — APPROVED** [review]

File: smarts/benchmark/driving_smarts/v2023/config.yaml (hunk 1 above)
Comment (Adaickalavan) [reply smarts/benchmark/driving_smarts/v2023/config.yaml]: I have appropriately updated the URLs.

File: smarts/benchmark/driving_smarts/v2022/config.yaml (hunk 2 above)
Comment (Adaickalavan) [reply smarts/benchmark/driving_smarts/v2022/config.yaml:42]: Agreed.

**Pull request merged**